x-pack/test/detection_engine_api_integration/security_and_spaces/tests/create_index.ts

```diff
@@ -0,0 +1,425 @@
+
+      it('should be able to create a signal index when it has not been created yet', async () => {
+        const { body } = await supertest
+          .post(DETECTION_ENGINE_INDEX_URL)
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(200);
+        expect(body).to.eql({ acknowledged: true });
+      });
+
+      it('should be able to create a signal index two times in a row as the REST call is idempotent', async () => {
+        await supertest.post(DETECTION_ENGINE_INDEX_URL).set('kbn-xsrf', 'true').send().expect(200);
+        const { body } = await supertest
+          .post(DETECTION_ENGINE_INDEX_URL)
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(200);
+        expect(body).to.eql({ acknowledged: true });
+      });
+
+      it('should be able to read the index name and status as not being outdated', async () => {
+        await supertest.post(DETECTION_ENGINE_INDEX_URL).set('kbn-xsrf', 'true').send().expect(200);
+
+        const { body } = await supertest.get(DETECTION_ENGINE_INDEX_URL).send().expect(200);
+        expect(body).to.eql({
+          index_mapping_outdated: false,
+          name: `${DEFAULT_SIGNALS_INDEX}-default`,
+        });
+      });
+    });
+
+    describe('t1_analyst', () => {
+      const role = ROLES.t1_analyst;
+      beforeEach(async () => {
+        await createUserAndRole(security, role);
+      });
+
+      it('should return a 404 when the signal index has never been created', async () => {
+        const { body } = await supertestWithoutAuth
+          .get(DETECTION_ENGINE_INDEX_URL)
+          .auth(role, 'changeme')
+          .send()
+          .expect(404);
+        expect(body).to.eql({ message: 'index for this space does not exist', status_code: 404 });
+      });
+
+      it('should NOT be able to create a signal index when it has not been created yet. Should return a 403 and error that the user is unauthorized', async () => {
+        const { body } = await supertestWithoutAuth
+          .post(DETECTION_ENGINE_INDEX_URL)
+          .set('kbn-xsrf', 'true')
+          .auth(role, 'changeme')
+          .send()
+          .expect(403);
+        expect(body).to.eql({
+          message:
+            'security_exception: action [cluster:admin/ilm/get] is unauthorized for user [t1_analyst], this action is granted by the privileges [read_ilm,manage_ilm,manage,all]',
+          status_code: 403,
+        });
+      });
+
+      it('should be able to read the index name and status as not being outdated', async () => {
+        // create the index using super user since this user cannot create the index
+        await supertest.post(DETECTION_ENGINE_INDEX_URL).set('kbn-xsrf', 'true').send().expect(200);
+
+        const { body } = await supertestWithoutAuth
+          .get(DETECTION_ENGINE_INDEX_URL)
+          .auth(role, 'changeme')
+          .send()
+          .expect(200);
+        expect(body).to.eql({
+          index_mapping_outdated: null,
```

Comment:
NOTE: I think this `null` is returned intentionally for when the user does not have correct permissions to read if the index mapping exists or not. It exists here in a few other tests.